jenkinshelper.ksh

```diff
@@ -0,0 +1,139 @@
+#!/bin/ksh
+# 2021-04-07 Olaf Bohlen <olbohlen@eenfach.de>
+# instead of putting all this into the Jenkinsfile I decided to put the actual code in this script
+
+# global config
+HTTPCONF="/etc/apache2/2.4/conf.d/pkgdepotd.conf"
+
+# just run prepare once to initially set up the environment
+prepare() {
+    echo "jenkinshelper: preparing..."
+    pkg install web/server/apache-24
+    mkdir -p /etc/apache2/2.4/conf.d && chown root:sys /etc/apache2/2.4/conf.d
+    grep "#ProxyPassMatch /default/(.*)\$ http://127.0.0.1:10000/\$1 nocanon max=200" "${HTTPCONF}" >/dev/null 2>&1
+    if [ $? -gt 0 ]; then
+	echo "jenkinshelper: Initializing a new apache config at ${HTTPCONF}"
+	echo "#ProxyPassMatch /default/(.*)\$ http://127.0.0.1:10000/\$1 nocanon max=200" >"${HTTPCONF}"
+    else
+	echo "jenkinshelper: Preserving an existing ${HTTPCONF}"
+    fi
+
+    cat >/etc/apache2/2.4/conf.d/00-proxy.conf <<EOF
+LoadModule proxy_module libexec/mod_proxy.so
+LoadModule proxy_connect_module libexec/mod_proxy_connect.so
+LoadModule proxy_ftp_module libexec/mod_proxy_ftp.so
+LoadModule proxy_http_module libexec/mod_proxy_http.so
+LoadModule proxy_ajp_module libexec/mod_proxy_ajp.so
+
+ProxyRequests Off
+ProxyVia Off
+ProxyPreserveHost on
+RequestHeader unset Origin
+AllowEncodedSlashes NoDecode
+EOF
+    
+    chown root:builders "${HTTPCONF}"
```

Comment:
This group does not exist initially, right?

On a related note, I had a somewhat suspended work (that can move forward since a week ago, that pkg(5) PR contributing to that got merged), to add a standard account in our userland mappings, `abuild:399` (GID and UID) to match "automated builder" originally of OBS, but also our dayjob CI farm uses that to run Jenkins builds (in an account separate from the controller's `jenkins` account) since the two CI-related systems toss the ball a lot in our case.

So I either had a PR or planned to have one, to have such user defined in jenkins-common and to deliver finally the jenkins-swarm as a package to get automated self-registering builders running as local `abuild` in their zones or wherever.

So TL;DR: it may make sense to cooperate here and define a new special account once in our packaging, and in a more standardized manner at that :-)

Reply:
This is specific to the jenkins setup, which started before the PR about abuild landed. builders is the group that all users that login to the build zone to build and run their work have. This could be made related to abuild group but it's definetly future infra work. We should start doing tasks for these.

Reply:
Ah, might as well rename that there in the Jenkins setup =p

One semi-useful point is that since this PR and recipe lands into common oi-userland, it might as well be usable on developers' machines with (a zone for) local Jenkins building the same. And so using bits delivered by OI packaging.

Reply:
Yes but I don't want to depend on abuild right now. Let's change this when we add abuild to jenkins dependencies.